AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_processor_cache_server.py

```diff
@@ -0,0 +1,178 @@
+from ly_test_tools.o3de.asset_processor import StopReason
+
+# Import fixtures
+from ..ap_fixtures.asset_processor_fixture import asset_processor as asset_processor
+from ..ap_fixtures.ap_setup_fixture import ap_setup_fixture as ap_setup_fixture
+from ..ap_fixtures.ap_idle_fixture import TimestampChecker
+
+# Use the following logging pattern to hook all test logging together:
+logger = logging.getLogger(__name__)
+
+# Helper: variables we will use for parameter values in the test:
+targetProjects = ["AutomatedTesting"]
+
+@pytest.fixture
+def local_resources(request, workspace, ap_setup_fixture):
+    # Test-level asset folder. Directory contains a sub-folder for each test (i.e. C1234567)
+    ap_setup_fixture["tests_dir"] = os.path.dirname(os.path.realpath(__file__))
+
+@pytest.fixture
+def ap_idle(workspace, ap_setup_fixture):
+    gui_timeout_max = 30
+    return TimestampChecker(workspace.paths.asset_catalog(ASSET_PROCESSOR_PLATFORM_MAP[workspace.asset_processor_platform]), gui_timeout_max)
+
+@pytest.mark.usefixtures("asset_processor")
+@pytest.mark.usefixtures("ap_setup_fixture")
+@pytest.mark.usefixtures("local_resources")
+@pytest.mark.parametrize("project", targetProjects)
+@pytest.mark.SUITE_periodic
+@pytest.mark.assetpipeline
+class TestsAssetProcessorGUI_Windows(object):
+    """
+    Tests for Asset Processor's Asset Cache Server (aka Shared Cache) feature
+    """
+
+    def get_file_hash(self, filename):
+        """
+        Utility function to hash the contents of a file
+        """
+        BUF_SIZE = 1024
+        sha1 = hashlib.sha1()
+
+        with open(filename, 'rb') as f:
+            while True:
+                data = f.read(BUF_SIZE)
+                if not data:
+                    return sha1.hexdigest()
+                sha1.update(data)
+
+    def find_archive_parts(self, path, prefix):
+        """
+        Finds the files that start with a certain prefix inside a path folder
+        """
+        result = {}
+        for _, _, files in os.walk(path):
+            for name in files:
+                if name.startswith(prefix):
+                    fingerprint = name[(len(prefix)):]
+                    result[fingerprint] = name
+        return result, files
+
+    def cycle_asset_processor(self, asset_processor, cache_mode, cache_folder, cache_pattern_name):
+        """
+        This launches the Asset Processor with the Asset Cache Server mode either enabled or inactive
+        """
+        extra_parameters = [
+            f' --regset="/O3DE/AssetProcessor/Settings/Server/cacheServerAddress={cache_folder}" ',
+            f' --regset="/O3DE/AssetProcessor/Settings/Server/assetCacheServerMode={cache_mode}" ',
+            f' --regset="/O3DE/AssetProcessor/Settings/Server/ACS {cache_pattern_name}/checkServer=true" '
+        ]
+
+        # execute in cache mode
+        result, _ = asset_processor.gui_process(quitonidle=False, extra_params=extra_parameters)
```

Comment:
Does this need to use the GUI?  Can it use batch mode?

Reply:
this only works in GUI right now since the server is meant to be run as an on going service (the batch runs then quits... kind of a bad server)
the client mode should work fine in batch mode, but that is not being tested here

Reply:
It wouldn't be too tough to add a flag to keep AP batch running once it hits idle.

Update this https://github.com/o3de/o3de/blob/development/Code/Tools/AssetProcessor/native/utilities/BatchApplicationManager.h#L46 to check for a flag, basically the opposite of https://github.com/o3de/o3de/blob/development/Code/Tools/AssetProcessor/native/utilities/GUIApplicationManager.cpp#L836

We might want to cut a ticket to do this later